### PR TITLE
Provide summarized acceptance test chunking upload steps

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
Make summarized steps for acceptance test chunking uploads that let you do the whole process by writing a single step.

## Related Issue
#31616 

## Motivation and Context
It is tedious to write all the individual chunking setup, upload and finalization acceptance test steps.
For most "happy" cases it is the same flow, so make summarized step definitions that can do old/new chunking with a single gherkin step.

## How Has This Been Tested?
Local test runs

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test enhancement

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by phil-davis** [comment]
Backport ``stable10`` #31621

**Comment by patrickjahns** [comment]
I gave this a thought today - it would probably also be great to have a chunking step that you could pass a file. The method would then split the file automatically and upload it as if it would have been via chunks? 

What do you think @IljaN @individual-it @phil-davis ?

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -1530,6 +1546,44 @@ public function userUploadsChunkedFile(
...
+		);
+	}
+
+	/**
+	 * New style chunking upload
+	 *
+	 * @param string $user
+	 * @param string $file
+	 * @param string $chunkingId
+	 * @param array $chunkDetails
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebDav.php]: Some documentation about what the array should contain would be helpful

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -1501,6 +1496,27 @@ public function userCreatesAFolder($user, $destination) {
...
+	/**
+	 * Old style chunking upload
+	 *
+	 * @When user :user uploads the following :total chunks to :file with old chunking and using the API
+	 * @Given user :user has uploaded the following :total chunks to :file with old chunking and using the API
+	 *
+	 * @param string $user
+	 * @param string $total
+	 * @param string $file
+	 * @param TableNode $chunkDetails
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebDav.php]: Documentation about what the table should contain would be useful

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 3]
```diff
@@ -1530,6 +1546,44 @@ public function userUploadsChunkedFile(
...
 
+	/**
+	 * New style chunking upload
+	 *
+	 * @When user :user uploads the following chunks to :file with new chunking and using the API
+	 * @Given user :user has uploaded the following chunks to :file with new chunking and using the API
+	 *
+	 * @param string $user
+	 * @param string $file
+	 * @param TableNode $chunkDetails
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebDav.php]: More documentation details

**Review by individual-it — CHANGES_REQUESTED** [review]
Code looks good, I would prefer to have a bit more documentation

**Comment by phil-davis** [comment]
PHP doc added.

**Review by individual-it — APPROVED** [review]

**Comment by individual-it** [comment]
Drone failed because of docker problems, restarted

**Pull request merged**